std/typecons.d

```diff
@@ -8725,28 +8727,50 @@ placeholder type `This` in $(REF Algebraic, std,variant).
 Returns: `ReplaceType` aliases itself to the type(s) that result after
 replacement.
 */
-template ReplaceType(From, To, T...)
+alias ReplaceType(From, To, T...) = ReplaceTypeUnless!(false_, From, To, T);
```

Comment:
Adding new public templates will make the approval process much slower (potentially taking weeks longer). I suggest keeping `ReplaceType` and `ReplaceTypeUnless` internal-use-only in this PR and making them in a future PR if you want.

Reply:
I'd really like `ReplaceTypeUnless` to be public so that I can use it in `sumtype` too without copy+pasting. I'm happy to split this into two PRs if it will make the overall process easier or more efficient, but that's a question of "how" best to make `ReplaceTypeUnless` public, not "if".

Reply:
Seems like `ReplaceType` would be a candidate for `std.traits`.

Reply:
If I were starting from scratch, I'd put it in `std.meta`, alongside non-recursive `Replace` and `ReplaceAll`. Given that `ReplaceType` is already in `std.typecons`, however, I don't think it makes sense to put `ReplaceTypeUnless` anywhere else.

Reply:
That makes sense.